### PR TITLE
Fix openapi authorization header for type token

### PR DESCRIPTION
**Pull request by calvernaz**
small fix to handle openapi `x-amazon-apigateway-authorizer` extension for authorizer type `TOKEN`.

https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authorizer.html

### CODE REVIEW

File: localstack/services/apigateway/helpers.py [hunk 1]
```diff
@@ -589,7 +590,12 @@ def create_authorizer(path_payload: dict) -> Optional[Authorizer]:
                         "authorizerCredentials"
                     ):
                         authorizer["authorizerCredentials"] = authorizer_credentials
-                    if identity_source := aws_apigateway_authorizer.get("identitySource"):
+                    if authorizer_type == "TOKEN":
+                        header_name = security_config.get("name")
+                        authorizer["identitySource"] = f"method.request.header.{header_name}"
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:595]: Not sure if we can rely on the `name` of the security config to be the header name of the  `identitySource` here? I understand that this is part of the examples in the [AWS docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authorizer.html#api-gateway-swagger-extensions-authorizer-example) , but seems to me that the `name` can be arbitrarily chosen (and hence may not necessarily map to the header name).

Maybe the default  `identitySource` should rather be `"method.request.header.Authorization"` here? As usual, a parity/snapshot test would help us get ultimate clarity on this question. 🤷

**Review by whummer — COMMENTED** [review]
+1 on the refined logic based on `authorizer_type` 👍 - just not sure if we're selecting the default `identitySource` header name properly here..

File: localstack/services/apigateway/helpers.py (hunk 1 above)
Comment (calvernaz) [reply localstack/services/apigateway/helpers.py:595]: not sure I agree here, so the `name` is a specific field of the authorizer extension, and for the type `TOKEN` must be a header. 

> Maybe the default  identitySource should rather be "method.request.header.Authorization" here? As usual, a parity/snapshot test would help us get ultimate clarity on this question.

that's precisely one of the issues, we had always default to that and must use whatever the user defines - https://github.com/localstack/localstack-ext/blob/289bfe82cfa11f9b79412938e8c647c69aa2375c/localstack_ext/services/apigateway/authorizers.py#L387-L389

Comment (whummer) [reply localstack/services/apigateway/helpers.py:595]: Actually, I think you're right - thanks for clarifying! Looking more closely at this line in the [docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authorizer.html#api-gateway-swagger-extensions-authorizer-example):


```
      "name" : "Authorization",                  // The name of the header containing the authorization token.
```

**Review by whummer — APPROVED** [review]

**Pull request merged**